pkg/agent/agent.go

```diff
@@ -335,7 +335,7 @@ func (a *Agent) setAPIKeyConfigFrom(ctx context.Context, secrets []*kates.Secret
 	// there _should_ only be one secret here, but we're going to loop and check that the object
 	// meta matches what we expect
 	for _, secret := range secrets {
-		if (secret.GetName() == a.agentCloudResourceConfigName || secret.GetName() == cloudConnectTokenDefaultResource) && secret.GetNamespace() == a.agentNamespace {
+		if (secret.GetName() == a.agentCloudResourceConfigName || strings.HasSuffix(secret.GetName(), cloudConnectTokenDefaultSuffix)) && secret.GetNamespace() == a.agentNamespace {
 			connTokenBytes, ok := secret.Data[cloudConnectTokenKey]
 			connToken := string(connTokenBytes)
 			dlog.Infof(ctx, "Setting cloud connect token from secret")
```

Comment:
And then this line could say `"Setting cloud connect token from 'ambassador-agent-cloud-token.default' secret"`

Reply:
yep